### ISSUE TITLE
Publish to pypi

### ISSUE THREAD

**Issue by engeir** [uit-cosmo/superposed-pulses#6]
Might consider publishing to pypi for easier use as a dependency in other projects, and as a "no-dev" option?

For further context, my issue is the following: I have a project (e.g. [volcano-cooking](https://github.com/engeir/volcano-cooking)), and the package itself and dependencies are installed with some system (in my case [poetry](https://python-poetry.org)). In this case I need to specify in a note that `fppy` must be installed manually. (Using `pip install git+<repo-url>` does not work, see drop down below).

<details><summary><i><b>Install of fppy</b></i></summary><br><ul>

```console
een023 in ~ via 🐍 v3.8.12 (test-fppy) took 4s (10:17:58)
➜  python
Python 3.8.12 (default, Nov  8 2021, 16:42:44)
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'numpy'
>>>

een023 in ~ via 🐍 v3.8.12 (test-fppy) took 3s (10:18:04)
➜  python -m pip install git+https://github.com/uit-cosmo/fppy.git
Collecting git+https://github.com/uit-cosmo/fppy.git
  Cloning https://github.com/uit-cosmo/fppy.git to /tmp/pip-req-build-etjsh6vz
  Running command git clone -q https://github.com/uit-cosmo/fppy.git /tmp/pip-req-build-etjsh6vz
Collecting numpy>=1.15.0
  Using cached numpy-1.22.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.8 MB)
Collecting scipy>=1.4.0
  Downloading scipy-1.8.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (41.6 MB)
     |████████████████████████████████| 41.6 MB 80 kB/s
Collecting tqdm>=4.50.2
  Using cached tqdm-4.62.3-py2.py3-none-any.whl (76 kB)
Collecting matplotlib>=3.2.0
  Using cached matplotlib-3.5.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (11.3 MB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting packaging>=20.0
  Using cached packaging-21.3-py3-none-any.whl (40 kB)
Collecting pyparsing>=2.2.1
  Using cached pyparsing-3.0.7-py3-none-any.whl (98 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.3.2-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.whl (1.2 MB)
Collecting python-dateutil>=2.7
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting pillow>=6.2.0
  Downloading Pillow-9.0.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.3 MB)
     |████████████████████████████████| 4.3 MB 4.8 MB/s
Collecting fonttools>=4.22.0
  Downloading fonttools-4.29.1-py3-none-any.whl (895 kB)
     |████████████████████████████████| 895 kB 1.9 MB/s
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Using legacy 'setup.py install' for fppy, since package 'wheel' is not installed.
Installing collected packages: six, pyparsing, python-dateutil, pillow, packaging, numpy, kiwisolver, fonttools, cycler, tqdm, scipy, matplotlib, fppy
    Running setup.py install for fppy ... done
Successfully installed cycler-0.11.0 fonttools-4.29.1 fppy-1.0 kiwisolver-1.3.2 matplotlib-3.5.1 numpy-1.22.2 packaging-21.3 pillow-9.0.1 pyparsing-3.0.7 python-dateutil-2.8.2 scipy-1.8.0 six-1.16.0 tqdm-4.62.3
WARNING: You are using pip version 21.1.1; however, version 22.0.3 is available.
You should consider upgrading via the '/home/een023/.pyenv/versions/test-fppy/bin/python -m pip install --upgrade pip' command.

een023 in ~ via 🐍 v3.8.12 (test-fppy) took 27s (10:18:37)
➜  python
Python 3.8.12 (default, Nov  8 2021, 16:42:44)
[GCC 10.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy
>>> import fppy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'fppy'
>>>

een023 in ~ via 🐍 v3.8.12 (test-fppy) took 6s (10:18:48)
➜
```

</ul></details>

Perhaps it is also useful to migrate from using `setup.py` to [`pyproject.toml`](https://pip.pypa.io/en/stable/reference/build-system/setup-py/)?

**Comment by Sosnowsky** [comment]
3ppy is now uploaded to pypi.
Didn't manage to make it work with pyproject.toml, so I kept the same setup file. Might look into this later.

**Comment by engeir** [comment]
It seems like the source code went missing? The [archive](https://files.pythonhosted.org/packages/ad/1f/11aab5f9dd99e09c44ab78a352ef6a6045b9926dc790cc51b815a02e03eb/3ppy-1.0.tar.gz) on [PyPI](https://pypi.org/project/3ppy/#files) contain only install instructions, but no folder:

```console
$ pip install 3ppy
...
ERROR: Could not find a version that satisfies the requirement 3ppy (from versions: 1.0)
ERROR: No matching distribution found for 3ppy
...
```

**Comment by Sosnowsky** [comment]
Ops, should be fine now!

**Comment by engeir** [comment]
Resolved by #9
